vision/detect/detect_test.go

```diff
@@ -66,3 +72,62 @@ func TestDetect(t *testing.T) {
 		}
 	}
 }
+
+func TestDetectAsyncDocument(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	ctx := context.Background()
+
+	// Create a temporary bucket
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bucketName := fmt.Sprintf("%s-golang-samples-%d", tc.ProjectID, time.Now().Unix())
+	bucket := client.Bucket(bucketName)
+	if err := bucket.Create(ctx, tc.ProjectID, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	// Run the test
+	var buf bytes.Buffer
+	gcs_source_uri := "gs://python-docs-samples-tests/HodgeConj.pdf"
+	gcs_destination_uri := "gs://" + bucketName + "/vision/"
+	err = detectAsyncDocument(&buf, gcs_source_uri, gcs_destination_uri)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check that the output files exist
+	_, err = bucket.Object("vision/output-1-to-2.json").Attrs(ctx)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
If this test fails, I'm concerned it won't be obvious why. Maybe there could be a list of expected objects then a for loop to check them all?

```go
files := []string{
	"vision/output-1-to-2.json",
	...
}
for _, want := range files {
	_, err = bucket.Object(want).Attrs(ctx)
	if err != nil {
		t.Fatalf("wanted object %q, got error: %v", err)
	}
}
```